src/app/core/services/cookies/cookies.service.ts

```diff
@@ -44,4 +32,53 @@ export class CookiesService {
   put(key: string, value: string, options?: CookiesOptions) {
     this.cookiesService.put(key, value, options);
   }
+
+  /**
+   * Saves the given cookie consent options settings together with the current cookie consent version
+   * to a users cookie named 'cookieConsent' and reloads the PWA application with the new settings.
+   * @param options The selected cookie consent options that should be enabled.
+   */
+  setCookiesConsentFor(options: string[]) {
+    const cookieConsentVersion = this.transferState.get<number>(COOKIE_CONSENT_VERSION, 1);
+    this.deleteAllCookies();
+    this.put('cookieConsent', JSON.stringify({ enabledOptions: options, version: cookieConsentVersion }), {
+      expires: new Date(new Date().setFullYear(new Date().getFullYear() + 1)),
+    });
+    window.location.reload();
+  }
+
+  /**
+   * Saves all configured cookie consent options as the users cookie consent settings.
+   */
+  setCookiesConsentForAll() {
+    this.setCookiesConsentFor(this.cookieConsentOptions?.options.map(x => x.id));
+  }
+
+  /**
+   * Determines the current users cookie consent setting for the requested cookie consent option.
```

Comment:
.... "check if consent was given for {option}"

Reply:
I still think this should be an Observable... 🤔

Reply:
What about the Observable?
Currently I tend to the simple method for checking since it seems easier to include this check in different places not needing to be reactive. With the Cookies Consent change page reload we also do not need it as Observable so far. The question might be if the reload is a good solution?

Reply:
Let's keep it simple the way it is. 80/20 case 😉